src/PandoraMonitoring.cc

```diff
@@ -1207,6 +1326,182 @@ void PandoraMonitoring::InitializeEve(Char_t transparency)
+
+    pTEveWindowPack = pTEveWindowPack->NewSlot()->MakePack();
+    pTEveWindowPack->SetShowTitleBar(kFALSE);
+    pTEveWindowPack->NewSlot()->MakeCurrent();
+    m_p2DUView = m_pEveManager->SpawnNewViewer("2D U View", "");
+    this->AddScenes(m_p2DUView, m_p2DUEventScene, m_p2DUGeometryScene, TGLViewer::kCameraOrthoXOZ, 1);
+
+    pTEveWindowPack->NewSlot()->MakeCurrent();
+    pTEveWindowPack->SetShowTitleBar(kFALSE);
+    m_p2DVView = m_pEveManager->SpawnNewViewer("2D V View", "");
+    this->AddScenes(m_p2DVView, m_p2DVEventScene, m_p2DVGeometryScene, TGLViewer::kCameraOrthoXOZ, 1);
+
+    pTEveWindowPack->NewSlot()->MakeCurrent();
+    pTEveWindowPack->SetShowTitleBar(kFALSE);
+    m_p2DWView = m_pEveManager->SpawnNewViewer("2D W View", "");
+    this->AddScenes(m_p2DWView, m_p2DWEventScene, m_p2DWGeometryScene, TGLViewer::kCameraOrthoXOZ, 1);
+
+    // Individual 2D and 3D views
+    pTEveWindowSlot = TEveWindow::CreateWindowInTab(m_pEveManager->GetBrowser()->GetTabRight());
+    pTEveWindowPack = pTEveWindowSlot->MakePack();
+    pTEveWindowPack->SetElementName("3D View");
+    pTEveWindowPack->SetHorizontal();
+    pTEveWindowPack->SetShowTitleBar(kFALSE);
+    pTEveWindowPack->NewSlot()->MakeCurrent();
+    m_p3DView = m_pEveManager->SpawnNewViewer("3D View", "");
+    this->AddScenes(m_p3DView, m_p3DEventScene, m_p3DGeometryScene, TGLViewer::kCameraPerspXOZ, 1);
+
+    pTEveWindowSlot = TEveWindow::CreateWindowInTab(m_pEveManager->GetBrowser()->GetTabRight());
+    pTEveWindowPack = pTEveWindowSlot->MakePack();
+    pTEveWindowPack->SetElementName("W View");
+    pTEveWindowPack->SetHorizontal();
+    pTEveWindowPack->SetShowTitleBar(kFALSE);
+    pTEveWindowPack->NewSlot()->MakeCurrent();
+    m_p2DWView = m_pEveManager->SpawnNewViewer("2D W View", "");
+    this->AddScenes(m_p2DWView, m_p2DWEventScene, m_p2DWGeometryScene, TGLViewer::kCameraOrthoXOZ, 1);
+  
+    pTEveWindowSlot = TEveWindow::CreateWindowInTab(m_pEveManager->GetBrowser()->GetTabRight());
+    pTEveWindowPack = pTEveWindowSlot->MakePack();
+    pTEveWindowPack->SetElementName("U View");
+    pTEveWindowPack->SetHorizontal();
+    pTEveWindowPack->SetShowTitleBar(kFALSE);
+    pTEveWindowPack->NewSlot()->MakeCurrent();
+    m_p2DUView = m_pEveManager->SpawnNewViewer("2D U View", "");
+    this->AddScenes(m_p2DUView, m_p2DUEventScene, m_p2DUGeometryScene, TGLViewer::kCameraOrthoXOZ, 1);
+ 
+    pTEveWindowSlot = TEveWindow::CreateWindowInTab(m_pEveManager->GetBrowser()->GetTabRight());
+    pTEveWindowPack = pTEveWindowSlot->MakePack();
+    pTEveWindowPack->SetElementName("V View");
+    pTEveWindowPack->SetHorizontal();
+    pTEveWindowPack->SetShowTitleBar(kFALSE);
+    pTEveWindowPack->NewSlot()->MakeCurrent();
+    m_p2DVView = m_pEveManager->SpawnNewViewer("2D V View", "");
+    this->AddScenes(m_p2DVView, m_p2DVEventScene, m_p2DVGeometryScene, TGLViewer::kCameraOrthoXOZ, 1);
+
```

Comment:
In the long term, maybe there's some refactoring possible here?

Reply:
Possibly, this is quite ugly, but mostly different orderings/configurations seem to position the windows and tabs, which are all subtly different.